tests/functions/test_expand_grid.py

```diff
@@ -0,0 +1,318 @@
+        assert expand_grid(others=data)
+
+
+def test_scalar_to_list():
+    """
+    Test that dictionary values are all converted to lists.
+    """
+    data = {
+        "x": 1,
+        "y": "string",
+        "z": set((2, 3, 4)),
+        "a": tuple((26, 50)),
+        "b": None,
+        "c": 1.2,
+        "d": True,
+        "e": False,
+    }
+    expected = (
+        [],
+        {
+            "x": [1],
+            "y": ["string"],
+            "z": [2, 3, 4],
+            "a": [26, 50],
+            "b": [None],
+            "c": [1.2],
+            "d": [True],
+            "e": [False],
+        },
+    )
+    assert _check_instance(data) == expected
+
+
+def test_nested_dict():
+    """Raise error if dictionary is nested in a dictionary's values"""
+    data = {"x": {"y": 2}}
+    with pytest.raises(TypeError):
+        assert expand_grid(others=data)
+
+
+def test_numpy():
+    """Raise error if numpy array in dictionary's values is empty"""
+    data = {"x": np.array([])}
+    with pytest.raises(ValueError):
+        assert expand_grid(others=data)
+
+
+def test_numpy_1d():
+    """Test output from a 1d numpy array """
+    data = {"x": np.array([2, 3])}
+    expected = pd.DataFrame(np.array([2, 3]), columns=["x"])
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_numpy_2d():
+    """Test output from a 2d numpy array"""
+    data = {"x": np.array([[2, 3]])}
+    expected = pd.DataFrame(np.array([[2, 3]])).add_prefix("x_")
+    assert_frame_equal(expand_grid(others=data), expected)
+
+
+def test_numpy_gt_2d():
+    """Raise error if numpy array dimension is greater than 2"""
+    data = {"x": np.array([[[2, 3]]])}
+    with pytest.raises(TypeError):
+        assert expand_grid(others=data)
+
+
+def test_series_empty():
+    """Test that values in key value pair should not be empty ... for Series"""
+    data = {"x": pd.Series([], dtype="int")}
+    with pytest.raises(ValueError):
+        assert expand_grid(others=data)
+
+
+def test_series_not_multi_index_no_name():
+    """Test for single index series"""
+    data = {"x": pd.Series([2, 3])}
+    expected = pd.DataFrame([2, 3], columns=["x"])
+    assert_frame_equal(_check_instance(data)[0][0], expected)
+
+
+def test_series_not_multi_index_with_name():
+    """Test for single index series with name"""
+    data = {"x": pd.Series([2, 3], name="y")}
+    expected = pd.DataFrame([2, 3], columns=["x_y"])
+    assert_frame_equal(_check_instance(data)[0][0], expected)
+
+
+def test_series_multi_index():
+    """Test that multiIndexed series trigger error"""
+    data = {
+        "x": pd.Series([2, 3],
+                       index=pd.MultiIndex.from_arrays([[1, 2], [3, 4]]))
+    }
+    with pytest.raises(TypeError):
+        assert expand_grid(others=data)
+
+
+def test_dataframe_empty():
+    """Test for empty dataframes"""
+    data = {"x": pd.DataFrame([])}
+    with pytest.raises(ValueError):
+        assert expand_grid(others=data)
```

Comment:
Assertion not necessary.

```suggestion
        expand_grid(others=data)
```